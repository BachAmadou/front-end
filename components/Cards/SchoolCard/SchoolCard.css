.SchoolCard {
  max-width: 400px;
  height: 550px;
  font-family: var(--primaryFontFamily), sans-serif;
}

@media screen and (max-width: 576px) {
  .SchoolCard {
    height: 550px;
  }
}

.SchoolCard > img {
  height: 150px;
  margin: 1rem;
}

.backCard {
  background-color: var(--secondary);
  color: var(--white);
  position: relative;
}
