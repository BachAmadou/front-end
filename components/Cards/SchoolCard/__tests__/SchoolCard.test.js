import React from 'react';
import { mount } from 'enzyme';
import createShallowSnapshotTest from 'test-utils/createShallowSnapshotTest';
import SchoolCard from '../SchoolCard';

const locations = [
  {
    address1: '825 Battery Street',
    address2: '3rd Floor',
    city: 'San Francisco',
    state: 'CA',
    va_accepted: true,
    zip: 94111,
  },
  {
    address1: '123Battery Street',
    address2: 'Ste 1',
    city: 'San Diego',
    state: 'CA',
    va_accepted: false,
    zip: 90111,
  },
];

describe('SchoolCard', () => {
  let componentInstance;
  let wrapper;
  let onClickMock;
  beforeEach(() => {
    componentInstance = (
      <SchoolCard
        cardFlipCallback={onClickMock}
        hasHardwareIncluded
        hasHousing
        hasOnline
        hasOnlyOnline={false}
        isFullTime
        locations={locations}
        logoSource="source"
        name="school name"
        website="website"
      />
    );
    onClickMock = jest.fn();
    wrapper = mount(componentInstance);
  });

  it('should render with required props', () => {
    createShallowSnapshotTest(componentInstance);
  });

  it('should change state when see locations button is clicked', () => {
    expect(wrapper.state('isFrontOfCardShowing')).toBe(true);
    wrapper
      .find('button')
      .filterWhere(node => node.text() === 'See Locations')
      .simulate('click');
    expect(wrapper.state('isFrontOfCardShowing')).toBe(false);
  });

  it('should render the "See Locations" button when multiple locations exist', () => {
    // ensure wrapper has multiple locations at this point
    expect(wrapper.prop('locations').length).toBeGreaterThan(1);
    expect(wrapper.find('button').filterWhere(node => node.text() === 'See Locations')).toExist();
  });
});
