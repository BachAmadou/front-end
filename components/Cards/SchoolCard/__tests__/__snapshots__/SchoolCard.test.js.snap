// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolCard should render with required props 1`] = `
<Card
  className="SchoolCard"
  hasAnimationOnHover={false}
>
  <FrontSchoolCard
    cardFlipCallback={[Function]}
    hasHardwareIncluded={true}
    hasHousing={true}
    hasOnline={true}
    hasOnlyOnline={false}
    isFullTime={true}
    locations={
      Array [
        Object {
          "address1": "825 Battery Street",
          "address2": "3rd Floor",
          "city": "San Francisco",
          "state": "CA",
          "va_accepted": true,
          "zip": 94111,
        },
        Object {
          "address1": "123Battery Street",
          "address2": "Ste 1",
          "city": "San Diego",
          "state": "CA",
          "va_accepted": false,
          "zip": 90111,
        },
      ]
    }
    logoSource="source"
    name="school name"
    website="website"
  />
</Card>
`;
